### PR TITLE
 PHP 8.1: (get|add|update|delete)_option: fix null to non-nullable deprecations (Trac 53635)

### PR DESCRIPTION
**Pull request by jrfnl**
### Tests_Option_Option::test_bad_option_names(): rework to data provider

* Rework the test method to use a data provider with named data sets instead of a `foreach()`.
* Add method visibility modifiers.
* Add the `$message` parameter for each assertion.

### Tests_Option_Option: add tests for valid, but undesired option names

... to safeguard against regressions which a potential fix to the base issue may cause.

### PHP 8.1: (get|add|update|delete)_option: fix null to non-nullable deprecations

In all four of the `get_option()`, `add_option()`, `update_option()` and `delete_option()` functions, the `$option` parameter is passed to the PHP native `trim()` function without prior validation.

In PHP 8.1, this could lead to a `trim(): Passing null to parameter #1 ($string) of type string is deprecated` for each of these methods.

`trim()` expects a text string and is only useful when _passed_ a text string as no other variable type can contain whitespace.
At the same time, `trim()` will always return a `string`, which means that in practice for any non-string values passed, it would effectively function as a type cast to string.

Fixed now by verifying that the `$option` is a scalar before processing it with `trim()`. This maintains the "type cast" behaviour, while still preventing the PHP deprecation if the `$option` name would be `null`.

This issue is already covered by the existing `Tests_Option_Option::test_bad_option_names()` test group and the new `test_valid_but_undesired_option_names()` tests.

Trac ticket: https://core.trac.wordpress.org/ticket/53635

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/option.php [hunk 1]
```diff
@@ -77,7 +77,10 @@
 function get_option( $option, $default = false ) {
 	global $wpdb;
 
-	$option = trim( $option );
+	if ( is_string( $option ) ) {
```
Comment (hellofromtonya) [inline src/wp-includes/option.php:80]: Need to change to `is_scalar()` to ensure floats are casted to int to ensure no b/c break if a float is passed in as the option. Also added more tests during live working session. Update is coming.

Comment (jrfnl) [reply src/wp-includes/option.php]: PR has been updated with all changes discussed during the livestream. PR description has been updated to match.

Comment (jrfnl) [reply src/wp-includes/option.php]: Ready for commit AFAIC.

**Review by hellofromtonya — APPROVED** [review]
[Thoroughly reviewed during working session](https://youtu.be/BQbaDBDCOIA?t=6205):

- Does not change current behavior (see tests and https://3v4l.org/pHg7N) as scalars are still processed through `trim`
- Fixes PHP 8.1 deprecation notice
- Skips doing `trim()` and variable assignment when not a scalar
- Using `is_scalar()` as `trim()` will cast to a string
- Improves tests

Ready for commit 👍

**Comment by hellofromtonya** [comment]
Committed with changesets https://core.trac.wordpress.org/changeset/51817 and https://core.trac.wordpress.org/changeset/51818.

**Pull request closed without merging**